x/staking/spec/02_state_transitions.md

```diff
@@ -37,6 +37,9 @@ When a validator begins the unbonding process the following operations occur:
 - update the `Validator` object for this validator
 - insert a new record into the `ValidatorQueue` for this validator
 
+Thoughts: Validator could stop his node at time of unbonding and we could remove validator from validator set instantly.
+If it update status on epoching process, there could be some issues with reward processing. Need to discuss @sunny.
```

Comment:
Updating status from `bonded` to `unbonding` if due to slashing, will happen automatically